index.d.ts

```diff
@@ -115,16 +118,30 @@ declare namespace Cmd {
     args?: any[]
   ): MapCmd<A>;
 
-  export function run<A extends Action, B extends Action>(
-    f: Function,
+  type InferArgs<T> = T extends (...args: infer T) => any ? T : never;
```

Comment:
what exactly does infer do here?

Reply:
It basically says from this type this is the argument I want to infer my variable T to.

Its a conditional type that only matches the left hand size, its a bit of a quirky typescript syntax.

Simplified examples:
```
type NumberName<T> = T extends number ? 'number' : 'notNumber';
type N = NumberName<2>; // 'number' < literal
type S = NumberName<'a'>; // 'notNumber' < literal
```
Then add never:
```
type NumberName<T> = T extends number ? 'number' : never;
type N = NumberName<2>; // 'number' < literal
type S = NumberName<'a'>; // never;
```
Then add infer;
```
type Args<T> = T extends (...args: infer A) => any ? A : never;
type A = Args<(a: number, b?: string) => void>; // [number, (string | undefined)]
```

Hopefully that makes sense?

Reply:
Is this not the same as TypeScript's built-in `Parameters`?
See lib.es5.d.ts:
```
/**
 * Obtain the parameters of a function type in a tuple
 */
type Parameters<T extends (...args: any) => any> = T extends (...args: infer P) => any ? P : never;
```

Reply:
Yes, thanks. I've updated to use Parameters